### PR TITLE
Added datetime api endpoint

### PR DESCRIPTION
**Pull request by saru95**

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -104,6 +109,15 @@ module.exports = {
 
 };
 
+function getDateTimeJSON() {
+	datetime = {
```
Comment (avmnu-sng) [inline routes/index.js]: do return directly, no need to store to variable

File: tests/endpoints_tests.js [hunk 2]
```diff
@@ -26,6 +28,7 @@ var chaiHttp = require('chai-http');
 var server = require('../app');
 var should = chai.should();
 var range = require('lodash.range');
+var moment = require('moment');
```
Comment (avmnu-sng) [inline tests/endpoints_tests.js]: `const`

File: routes/index.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
 		var resource = req.params.resource,
 			itemArg = req.params[0] || req.query.id || null,
 			items;
-
-		if (data[resource] && !itemArg) {
+		
+		if (resource in nonStaticResourceMapping) {
+			items = nonStaticResourceMapping[resource];
+			return res.status(200).send({ data: items });
+		} else if (data[resource] && !itemArg) {
```
Comment (avmnu-sng) [inline routes/index.js]:
```javascript
if (resource in nonStaticResourceMapping) {
} else if (data[resource]) {
    if (itemArg) {
        return here
    }

    process and return for !itemArg here
}
```

File: tests/endpoints_tests.js [hunk 4]
```diff
@@ -638,4 +641,18 @@ describe('Check all `/api` endpoints', function() {
         });
     });
   });
+
+  describe('datetime', function() {
+    it('/api/datetime', function(done) {
```
Comment (avmnu-sng) [inline tests/endpoints_tests.js]: Use ECMAScript 6 syntax everywhere then

File: package.json [hunk 5]
```diff
@@ -13,7 +14,8 @@
     "cors": "^2.4.2",
     "express": "^4.9.7",
     "hbs": "^2.7.0",
-    "lodash": "^2.4.1"
+    "lodash": "^2.4.1",
```
Comment (akshay3004) [inline package.json]: where is lodash being used?

File: routes/index.js [hunk 6]
```diff
@@ -1,16 +1,21 @@
...
 	config = require("./../config.json"),
 	resources_config = require("./../resources_config.json"),
-	_ = require("lodash");
+	_ = require("lodash"),
+	moment = require('moment');
 
 module.exports = {
 
 	get: function(req, res, next) {
+		var nonStaticResourceMapping = { 'datetime': getDateTimeJSON() }
```
Comment (akshay3004) [inline routes/index.js]: Doesn't look like this is supposed to support most of the functionality given in this function. Can we route this call to a separate function instead?

File: package.json [hunk 7]
```diff
@@ -13,7 +14,8 @@
...
+    "moment": "^2.21.0"
```
Comment (akshay3004) [inline package.json]: moment js seems too bulky for this simple use case. Lets use a lightweight date library like:
https://github.com/date-fns/date-fns or https://github.com/taylorhakes/fecha

File: package.json (hunk 5 above)
Comment (saru95) [reply package.json]: It's already being used in `compile_data.js`.

Comment (akshay3004) [reply package.json]: My bad. I didn't realise that we just added a comma on this line

**Pull request merged**